packages/sdk/src/Incentives/ERC20PeggedIncentive.ts

```diff
@@ -0,0 +1,490 @@
+      this._config,
+      {
+        address: this.assertValidAddress(),
+        args: [prepareClaimPayload(payload)],
+        ...this.optionallyAttachAccount(),
+        // biome-ignore lint/suspicious/noExplicitAny: Accept any shape of valid wagmi/viem parameters, wagmi does the same thing internally
+        ...(params as any),
+      },
+    );
+    const hash = await writeErc20PeggedIncentiveClawback(this._config, request);
+    return { hash, result };
+  }
+
+  /**
+   * Check if an incentive is claimable.
+   *
+   * @public
+   * @async
+   * @param {ClaimPayload} payload
+   * @param {?ReadParams} [params]
+   * @returns {Promise<boolean>} = True if the incentive is claimable based on the data payload
+   */
+  public async isClaimable(payload: ClaimPayload, params?: ReadParams) {
+    return await readErc20PeggedIncentiveIsClaimable(this._config, {
+      address: this.assertValidAddress(),
+      args: [payload.target, payload.data],
+      // biome-ignore lint/suspicious/noExplicitAny: Accept any shape of valid wagmi/viem parameters, wagmi does the same thing internally
+      ...(params as any),
+    });
+  }
+
+  /**
+   * read the peg token for the incentive.
+   *
+   * @public
+   * @async
+   * @param {?ReadParams} [params]
+   * @returns {Promise<Address>} = The address of the token the reward is pegged to
+   */
+  public async getPeg(params?: ReadParams) {
+    return await readErc20PeggedIncentiveGetPeg(this._config, {
+      address: this.assertValidAddress(),
+      args: [],
+      // biome-ignore lint/suspicious/noExplicitAny: Accept any shape of valid wagmi/viem parameters, wagmi does the same thing internally
+      ...(params as any),
+    });
+  }
+
+  /**
+   * Get the maximum amount that can be claimed by this incentive. Useful when used in conjunction with `BoostCore.calculateProtocolFee`
+   *
+   * @public
+   * @async
+   * @param {?ReadParams} [params]
+   * @returns {Promise<bigint>} = Return a bigint representing that maximum amount that can be distributed by this incentive.
+   */
+  public async getTotalBudget(params?: ReadParams) {
+    return await this.limit(params);
+  }
+
+  /**
+   * Check if any claims remain by comparing the incentive's total claims against its limit. Does not take requesting user's elligibility into account.
+   *
+   * @public
+   * @async
+   * @param {?ReadParams} [params]
+   * @returns {Promise<boolean>} - True if total claims is less than limit
+   */
+  public async canBeClaimed(params?: ReadParams) {
+    return (await this.getRemainingClaimPotential(params)) > 0n;
+  }
+
+  /**
+   * Check how many claims remain by comparing the incentive's total claims against its limit. Does not take requesting user's elligibility into account.
+   *
+   * @public
+   * @async
+   * @param {?ReadParams} [params]
+   * @returns {Promise<bigint>} - True if total claims is less than limit
+   */
+  public async getRemainingClaimPotential(params?: ReadParams) {
```

Comment:
is getting the price conversion from the peg out of the scope of the sdk? It makes sense to be unopinionated about the conversion source, but just curious.

Reply:
Yeah I think so just cause I don't want to be opinionated about the price oracle and if it's an API vs contract source I don't like importing those dependencies.

Reply:
We also have a lot of the helpers/logic for price fees in the backend. We could add a helper that _given a price_ does the maths/conversion?

Reply:
@jonathandiep WDYT

Reply:
I think the helper would be nice, but it would only help for a short time knowing how volatile token prices are